lib/ansible/modules/cloud/azure/azure_rm_webapp_facts.py

```diff
@@ -0,0 +1,292 @@
+          state: Running
+          usageState: Normal
+        resource_group: ansiblewebapp1
+    }]
+'''
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.common import AzureMissingResourceHttpError, AzureHttpError
+except:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+AZURE_OBJECT_CLASS = 'WebApp'
+
+
+class AzureRMWebAppFacts(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            name=dict(type='str'),
+            resource_group=dict(type='str'),
+            tags=dict(type='list')
+        )
+
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_webapps=[])
+        )
+
+        self.name = None
+        self.resource_group = None
+        self.tags = None
+        self.info_level = None
+
+        self.framework_names = ['net_framework', 'java', 'php', 'node', 'python', 'dotnetcore', 'ruby']
+
+        super(AzureRMWebAppFacts, self).__init__(self.module_arg_spec,
+                                                 supports_tags=False,
+                                                 facts_module=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if self.name:
+            self.results['ansible_facts']['azure_webapps'] = self.list_by_name()
+        elif self.resource_group:
+            self.results['ansible_facts']['azure_webapps'] = self.list_by_resource_group()
+        else:
+            self.results['ansible_facts']['azure_webapps'] = self.list_all()
+
+        return self.results
+
+    def list_by_name(self):
+        self.log('Get web app {0}'.format(self.name))
+        item = None
+        result = []
+
+        try:
+            item = self.web_client.web_apps.get(self.resource_group, self.name)
+            site_config = self.list_webapp_configuration(self.resource_group, self.name)
+            app_settings =  self.list_webapp_appsettings(self.resource_group, self.name)
+        except CloudError:
+            pass
+
+        if item and self.has_tags(item.tags, self.tags):
+            pip = self.serialize_obj(item, AZURE_OBJECT_CLASS)
+
+            curated_result = self.construct_curated_webapp(pip, site_config, app_settings)
+            result = [curated_result]
+
+        return result
+
+    def list_by_resource_group(self):
+        self.log('List web apps in resource groups {0}'.format(self.resource_group))
+        try:
+            response = list(self.web_client.web_apps.list_by_resource_group(self.resource_group))
+        except CloudError as exc:
+            self.fail("Error listing web apps in resource groups {0} - {1}".format(self.resource_group, str(exc)))
+
+        results = []
+        for item in response:
+            if self.has_tags(item.tags, self.tags):
```

Comment:
there could be a function to convert entire list, as this code is repeated below

Reply:
fixed